src/rnanorm/__main__.py

```diff
@@ -53,18 +55,25 @@ def validate_args(args):
     """Validate that input files exist when they are needed."""
     expression_path = args.expression
     gene_lengths_path = args.gene_lengths
+    gtf_path = args.gtf
     tpm_output_path = args.tpm_output
     cpm_output_path = args.cpm_output
 
     if not os.path.exists(expression_path):
         raise ArgumentValidateException(f"Expressions file not found {expression_path}")
 
     if tpm_output_path:
-        if gene_lengths_path is None:
-            raise ArgumentValidateException("--gene-lengths must be given for --tpm-output")
+        if gene_lengths_path is None and gtf_path is None:
+            raise ArgumentValidateException("--gene-lengths or --gtf must be given for --tpm-output")
 
-        if not os.path.exists(gene_lengths_path):
-            raise ArgumentValidateException(f"Gene lengths file not found {gene_lengths_path}")
+        if gene_lengths_path is not None and gtf_path is not None:
+            raise ArgumentValidateException("Only one of --gene-lengths or --gtf must be given")
+
+        missing_file_path = next(
```

Comment:
I would rather have two `if` statements. One for the gene lengths file and the other for annotation file. Keep it simple.